app/controllers/candidate_interface/application_choices_controller.rb

```diff
@@ -1,8 +1,22 @@
 module CandidateInterface
-  class ApplicationChoicesController < ::CandidateInterface::ContinuousApplicationsController
-    before_action SubmissionPermissionFilter
-    before_action :redirect_to_your_applications_if_submitted
+  class ApplicationChoicesController < ContinuousApplicationsController
+    before_action SubmissionPermissionFilter, only: %i[submit confirm_destroy destroy]
+    before_action :redirect_to_your_applications_if_submitted, only: %i[submit confirm_destroy destroy]
+    before_action :redirect_to_post_offer_dashboard_if_accepted_deferred_or_recruited, only: :index
```

Comment:
I didn't understand this part (but maybe because I didn't draw it my head). If the user is: recruited, pending conditions or deferred can they go manually (or any other means) to the actions of this controller?

Reply:
Can we add test coverage (requests specs?) trying to submit or enter on any application choice when the user is: recruited (or pending conditions/deferred)? And the user should be redirected to the post offer dashboard instead?

Reply:
This was copied over directly from `app/controllers/candidate_interface/continuous_applications_choices_controller.rb` I only added the `only: ` conditions
There is no change to functionality here

Reply:
Yes I agree, but I think may be there a bug or "security" on this code if I understood correctly.

Reply:
For us to talk next week:

1. Have a recruited or pending conditions applications
2. Try to access other actions in the controller that are not the index
3. It should redirect the user to post offer dashboard

I supsect the third step may not be happening if I understood the code correctly. Let's talk next week.

Reply:
Reading more the code I understand

this line:

```ruby
before_action :redirect_to_post_offer_dashboard_if_accepted_deferred_or_recruited, only: :index
```

because the line is there on app/controllers/candidate_interface/continuous_applications_controller.rb:

```ruby
module CandidateInterface
   class ContinuousApplicationsController < CandidateInterfaceController
     before_action :redirect_to_post_offer_dashboard_if_accepted_deferred_or_recruited
    # ...
   end
end
```

But either way I think we should have more coverage about this for app/controllers/candidate_interface/application_choices_controller.rb in relation to redirect_to_post_offer_dashboard_if_accepted_deferred_or_recruited

Reply:
How do you feel about this being merged as it is and then we can card up and fix issues separately? I feel like I don't want to add additional context to be reviewed in this PR?

Reply:
Sure :)